yarn-project/simulator/src/avm/avm_machine_state.ts

```diff
@@ -52,6 +54,28 @@ export class AvmMachineState {
     return new AvmMachineState(state.l1GasLeft, state.l2GasLeft, state.daGasLeft);
   }
 
+  /**
+   * Consumes the given gas.
+   * Should any of the gas dimensions get depleted, it sets all gas left to zero and triggers
+   * an exceptional halt by throwing an OutOfGasError.
+   */
+  public consumeGas(gasCost: Partial<GasCost>) {
+    // Assert there is enough gas on every dimension.
+    const outOfGasDimensions = GasDimensions.filter(
+      dimension => this[`${dimension}Left`] - (gasCost[dimension] ?? 0) < 0,
+    );
+    // If not, trigger an exceptional halt.
+    // See https://yp-aztec.netlify.app/docs/public-vm/execution#gas-checks-and-tracking
+    if (outOfGasDimensions.length > 0) {
+      this.exceptionalHalt();
```

Comment:
You can use `this.revert([])` ?

Reply:
Revert didn't set gas to zero, that's why I set up a different function. Still, I wonder whether this is needed at all, since the state seems to be discarded once an exception is thrown.